### PR TITLE
Adds cause of death enum

### PR DESCRIPTION
**Pull request by rlgreen91**
Adds migration to create cause of death enum.  Adds rake task to convert cause of death reference data on case table to enum.

Addresses Issue #3005 in part

In your PR did you:

  - [X] Include a description of the changes?
  - [X] Mention the issue the PR addresses?
  - [X] Include screenshots of any changes to the UI?
  - [X] Isolate any changes to gems (meaning that any new, updated, or removed gems and resulting code changes should be in their own PR)?
  - [X] Add and/or update specs for your code?

### CODE REVIEW

**Comment by rlgreen91** [comment]
@trystant I can't for the life of me figure out why Rubocop is ignoring that I disabled the class length cop.  Either way, once we finish converting the model to an enum, that flag is likely to go away as other lines are deleted.

File: app/models/case.rb [hunk 1]
```diff
@@ -2,9 +2,21 @@
 
 # This is the main model of the application. Each death is a case.
 # TODO: Lots & lots of refactoring
+# rubocop:disable Metrics/ClassLength
 class Case < ApplicationRecord
   # TODO: Clean up relationship section
```
Comment (trystant) [inline app/models/case.rb]: This todo is from 2016. I think that that comment can be removed.

File: db/migrate/20190824175709_add_cause_of_death_to_postgres.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class AddCauseOfDeathToPostgres < ActiveRecord::Migration[5.0]
+  def up
+    execute <<-SQL
+      CREATE TYPE cause_of_death AS ENUM ('choking', 'shooting', 'beating', 'taser', 'vehicular', 'medical neglect', 'response to medical emergency', 'suicide')
+    SQL
+
+    add_column :cases, :cause_of_death, :cause_of_death, index: true
+  end
+
+  def down
```
Comment (trystant) [inline db/migrate/20190824175709_add_cause_of_death_to_postgres.rb]: I like that this is completely reversible.

File: app/models/case.rb (hunk 1 above)
Comment (rlgreen91) [reply app/models/case.rb]: Ok, I'll remove both of them in the next PR touching this file.

**Pull request merged**